src/utilities/didDowngrade/didDowngrade.ts

```diff
@@ -27,15 +27,10 @@ async function getSignedTransaction(
   seed: Uint8Array,
   fullDid: IDidDetails['did'],
 ): Promise<DidTransaction> {
-  const fullDidDetails = await queryFullDetailsFromIdentifier(
-    DidUtils.parseDidUrl(fullDid).identifier,
-  );
-  if (!fullDidDetails) {
-    throw new Error(`Could not resolve DID ${fullDid}`);
-  }
+  const fullDidDetails = await getFullDidDetails(fullDid);
 
   const extrinsic = await DidChain.getDeleteDidExtrinsic(
-    await DidChain.queryEndpointsCounts(fullDidDetails.did),
+    await DidChain.queryEndpointsCounts(fullDidDetails.identifier),
```

Comment:
Yes, I understand your point. But the point of a chain file is to be a stateless proxy to the chain functions, and the chain functions have knowledge of a DID identifier, not of a DID. It might make sense to provide a utility function in the `FullDidDetails` class that wraps a chain call, but for as far as the chain file goes, it should not be mixed with DID URIs, I think.

Reply:
I am a big fan of [the Zen of Python](https://peps.python.org/pep-0020/) programming language, and one of its points is "Although practicality beats purity." :)

In any case, I do not even know why as the user of the API I must call `queryEndpointsCounts` to create a delete extrinsic. DidChain could remain pure, I don’t care, but in that case I would appreciate a method in DidUtils which takes care of glueing things together.

Reply:
This is a good point. I will make a ticket to make it optional in the deletion extrinsic.